syntax = "proto2";

package ZilliqaMessage;

import "ZilliqaMessage.proto";


message DefaultResponse
{
    required string result = 1 [default = "hello"];
}

message UIntResponse
{
    required uint32 result = 1;
}

message StringResponse
{
    required string result = 1;
}

message DoubleResponse
{
    required double result = 1 [default = 0];
}

message UInt64Response
{
    required uint64 result = 1;
}

message CreateTransactionResponse
{
    optional string error = 1;
    optional string info = 2;
    optional string tranid = 3;
    optional string contractaddress = 4;
}

message GetTransactionResponse
{
    optional string error = 1;
    optional ProtoTransaction tx = 2;
    optional string receipt = 3;
}

message GetDSBlockResponse
{
    optional string error = 1;
    optional ProtoDSBlock dsblock = 2;
}

message GetTxBlockResponse
{
    optional string error = 1;
    optional ProtoTxBlock txblock = 2;
}

message GetBalanceResponse
{
    optional string error = 1;
    optional string balance = 2;
    optional string nonce = 3;
}

message GetSmartContractStateResponse
{
    optional string error = 1;
    optional string storagejson = 2;
}

message GetSmartContractInitResponse
{
    optional string error = 1;
    optional string initjson = 2;
}

message GetSmartContractCodeResponse
{
    optional string error = 1;
    optional string smartcontractcode = 2;
}

message GetSmartContractResponse
{
    optional string error = 1;
    message ContractAddressState
    {
        required string address = 1;
        required string state = 2;
    }
    repeated ContractAddressState address = 2;
}

message TxBlockListingResponse
{
    optional string error = 1;
    optional int32 maxpages = 2;
}
